share/functions/fish_config.fish

```diff
@@ -205,17 +205,33 @@ function fish_config --description "Launch fish's web based configuration"
                         echo "Too many arguments" >&2
                         return 1
                     end
-                    if not set -q argv[1]
+                    if not set -q argv[1] && contains -- $cmd choose
                         echo "Too few arguments" >&2
                         return 1
                     end
 
+                    # If we are choosing a theme, save its name to a hidden global variable.
+                    # If we are persisting a theme, either get its name from $argv[1] or
+                    # from the last chosen theme.
+                    if contains -- $cmd choose
+                        set -g __fish_last_chosen_theme $argv[1]
```

Comment:
This isn't the right way to go about this.

The current theme is just the set of the current color variables, so running `fish_config theme save` should just persist the current variables (to universal variables)

Reply:
Basically, what you need to do is just not read any file if $argv is empty, and instead only do the saving part.

So an `if set -q argv[1]` on line 226 .

Reply:
In essence, `fish_config theme save` or `fish_config prompt save` is the "I'm happy with this" button. It just saves the current config.

So if you do `fish_config theme choose coolbeans`, and you decide that it's nice, and then you do `fish_config theme save` it should save the coolbeans theme, but if you decide to change a color or two and then `fish_config theme save` it should save the theme *with those alterations*.

Reply:
That makes sense, thanks.